datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -108,6 +108,7 @@ pub fn name_to_op(name: &str) -> Option<Operator> {
         "multiply" => Some(Operator::Multiply),
         "divide" => Some(Operator::Divide),
         "mod" => Some(Operator::Modulo),
```

Comment:
because `mod` is the same as `modulus` we need to mark it as deprecated somehow, but how?

Reply:
I'm not sure how to produce a deprecation warning for this programatically.

However, we can provide an easy migration path if we also update the producer code to use `modulus` so we can release a version of DataFusion that consumes `mod` and `modulus`, and produces only `modulus`. Then the version after than we remove `mod` and announce it as a breaking change in the release notes.